worker/lease/manager.go

```diff
@@ -589,7 +589,7 @@ func (manager *Manager) setNextTimeout(t time.Time) {
 
 	// Ensure we never walk the next check back without have performed a
 	// scheduled check *unless* we're just starting up.
-	if !manager.nextTimeout.IsZero() && t.After(manager.nextTimeout) {
```

Comment:
I don't think this change is quite right. In the (albeit unlikely) event that you get the same `t`, we do needless processing to calculate a next scheduled tick that is the same as the last one.